src/org/python/modules/struct.java

```diff
@@ -1096,11 +1097,36 @@ public static PyTuple unpack(String format, PyArray buffer) {
             throw StructError("unpack str size does not match format");
          return unpack(f, size, format, new ByteStream(string));
     }
-    
+
```

Comment:
Again, I think we should accept any byte-like object (``BufferProtocol``) here but I haven't looked at the other ``unpack`` methods. It's a pain we have to make a string copy just so we can put the bottom 8 bits of each ``char`` in the ``ByteStream``. (Do we?) If it doesn't provoke a cascade of changes elsewhere, could we teach ``ByteStream`` the buffer protocol so it gets the bytes directly?